### DISCUSSION TITLE
Dropout inside Netket

### DISCUSSION THREAD

**Discussion by Mobo140** [netket/netket#1789] [Q&A]
I wanted to implement RBM sparse inside Netket, which is known as "dropout", I found #1695, but I didn't find the answer to my question: is it possible to do it now or not possible yet?

**Comment by PhilipVinc** [comment]
Not with standard `MCState` , for the same reason as also `BatchNorm`: we do not pipe through the training arguments that would be needed to support this.

However, it would not be hard to pipe them through like in https://github.com/netket/netket/pull/1485 . Probably, this PR might be enough to use dropout in netket.

There's some subtleties, however, with regards to how the rng for dropout is seeded and natural gradient/sr (you should use the same rng seed)

I would be curious to know how dropout fares in NQS learning scenario with natural gradient. I would think that we do not have overfitting problems so dropout is not as important as in traditional ML, but... who knows.
